### PR TITLE
feat(docs): adds a proposal for modulith arch

### PR DESCRIPTION
**Pull request by andypf**
This PR adds a proposal document outlining the Modulith Dashboard Architecture (Code Name: `Carbon`) under docs/proposals. The document details the new architecture’s design, which combines the stability of a monolithic core with the flexibility of a modular system. It serves as a foundational reference for future development and planning.

### CODE REVIEW

File: docs/proposals/001_modulith_architecture.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+**Author(s):** UI Team  
+**Status:** Draft
+
+---
+
+## 1. Problem Statement
+
+To create a new **Modulith Dashboard**, we need a flexible and scalable architecture that supports the integration of third-party extensions. The architecture should combine the benefits of modularity with the stability of a monolithic system, allowing for easy maintenance, upgrades, and adaptability to future requirements.
+
+_Example:_
```
Comment (edda) [inline docs/proposals/001_modulith_architecture.md:13]: Leftover Example headline.
```suggestion

```

File: docs/proposals/001_modulith_architecture.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+> The new Modulith Dashboard must enable seamless updates, interchangeable components, and third-party integrations while maintaining performance and stability.
```
Comment (edda) [inline docs/proposals/001_modulith_architecture.md:15]:
```suggestion
The new Modulith Dashboard must enable seamless updates, interchangeable components, and third-party integrations while maintaining performance and stability.
```

**Review by edda — CHANGES_REQUESTED** [review]

**Review by edda — DISMISSED** [review]

**Comment by barsukov** [comment]
Please add .mdx as a .prettier ignore to surpass this annoying check.

**Comment by andypf** [comment]
> Please add .mdx as a .prettier ignore to surpass this annoying check.

it is valid format issue, already fixed

**Review by barsukov — APPROVED** [review]

**Pull request merged**